### PR TITLE
SynchronousMemcpy changed to SynchronousMemcpyD2H and SynchronousMemcpyH2D

### PR DESCRIPTION
**Pull request by siju-samuel**
Updated deprecated SynchronousMemcpy to SynchronousMemcpyD2H and SynchronousMemcpyH2D

### CODE REVIEW

File: tensorflow/core/common_runtime/gpu/gpu_debug_allocator.cc [hunk 1]
```diff
@@ -44,7 +44,8 @@ bool CheckMask(se::StreamExecutor* exec, void* ptr, int64* mask) {
   se::DeviceMemory<int64> gpu_ptr{se::DeviceMemoryBase{ptr, MASK_BYTES}};
   int64 tmp[MASK_WORDS];
 
-  if (!exec->SynchronousMemcpy(&tmp, gpu_ptr, MASK_BYTES)) {
+  Status result = exec->SynchronousMemcpyD2H(gpu_ptr, MASK_BYTES, tmp);
```
Comment (aaroey) [inline tensorflow/core/common_runtime/gpu/gpu_debug_allocator.cc]: Please also log `result` below, same for other changed lines.

**Comment by siju-samuel** [comment]
@aaroey Thanks for pointing it out, printing 'result' during error will help in easy identification of the issue. Its updated.

File: tensorflow/core/common_runtime/gpu/gpu_debug_allocator.cc [hunk 2]
```diff
@@ -46,7 +46,7 @@ bool CheckMask(se::StreamExecutor* exec, void* ptr, int64* mask) {
 
   Status result = exec->SynchronousMemcpyD2H(gpu_ptr, MASK_BYTES, tmp);
   if (!result.ok()) {
-    LOG(FATAL) << "Could not copy debug mask";
+    LOG(FATAL) << "Could not copy debug mask, " << result.error_message();
```
Comment (aaroey) [inline tensorflow/core/common_runtime/gpu/gpu_debug_allocator.cc]: Just `<< result;` is sufficient, `Status` class has an overload for operator `<<`.

**Comment by siju-samuel** [comment]
@aaroey, Thanks for the review, Changed result.error_message() to result

**Review by aaroey — APPROVED** [review]
Thanks for the change.

**Pull request merged**